api/http/handler/helm/handler.go

```diff
@@ -81,3 +91,25 @@ func (handler *Handler) GetEndpoint(r *http.Request) (*portainer.Endpoint, *http
 
 	return endpoint, nil
 }
+
+// getHelmClusterAccess obtains the core k8s cluster access details from request.
+// The cluster access includes the cluster server url, the user's bearer token and the tls certificate.
+// The cluster access is passed in as kube config CLI params to helm binary.
+func (handler *Handler) getHelmClusterAccess(r *http.Request) (*options.KubernetesClusterAccess, *httperror.HandlerError) {
+	endpoint, httperr := handler.getEndpoint(r)
+	if httperr != nil {
+		return nil, httperr
+	}
+
+	bearerToken, err := security.ExtractBearerToken(r)
+	if err != nil {
+		return nil, &httperror.HandlerError{http.StatusUnauthorized, "Unauthorized", err}
+	}
+
+	clusterAccess := handler.kubeConfigService.GetKubeConfigInternal(endpoint.ID, bearerToken)
+	return &options.KubernetesClusterAccess{
```

Comment:
why do we need another identical object type?

Reply:
The `options.KubernetesClusterAccess` is from an external helm library (https://github.com/portainer/libhelm)
The `kubernetesClusterAccess` returned from `handler.kubeConfigService.GetKubeConfigInternal(endpoint.ID, bearerToken)` is a private struct relevant to the `kubeConfigService`.

We don't want to the `kubeConfigService` to be tightly coupled with the helm binary/library - since it is possible the same kubeconfig generated by this service would be applicable to other kube based CLI tools (`kubectl` for example).

Reply:
:+1: